app/controllers/voucher_adjustments_controller.rb

```diff
@@ -15,7 +15,11 @@ def create
   end
 
   def destroy
-    @order.voucher_adjustments.find_by(id: params[:id])&.destroy
+    # An order can have more than one adjustment linked to one voucher
+    adjustment = @order.voucher_adjustments.find_by(id: params[:id])
+    if adjustment.present?
+      @order.voucher_adjustments.where(originator_id: adjustment.originator_id)&.destroy_all
```

Comment:
Part of me wants to suggest that we move the knowledge about what the originator_id is used for, to the `VoucherAdjustmentsService` (which already has this knowledge). But.. I'm probably overthinking it.